File: Backend/app/schemas/system_log.py
"""
시스템 로그 Pydantic 스키마
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel


class SystemLogResponse(BaseModel):
    id: int
    timestamp: datetime
    type: str
    level: str
    source: str
    message: str
    detail: Optional[str] = None

    model_config = ConfigDict(
        from_attributes=True,
        alias_generator=to_camel,
        populate_by_name=True,
    )


class SystemLogListResponse(BaseModel):
    items: list[SystemLogResponse]
    total: int
    page: int
    size: int

    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
    )

File: Frontend/src/types/systemLog.ts
export type LogType = 'CONNECTION' | 'MESSAGE' | 'ERROR' | 'SYSTEM'
export type LogLevel = 'info' | 'warn' | 'error'

export interface SystemLog {
  id: number
  timestamp: string
  type: LogType
  level: LogLevel
  source: string       // 'MQTT', 'Server', 'App'
  message: string      // 한 줄 요약
  detail: string | null // JSON 상세 (행 확장 시 표시)
}

export interface SystemLogListResponse {
  items: SystemLog[]
  total: number
  page: number
  size: number
}
